tests/test_http.py

```diff
@@ -90,6 +90,21 @@ def test_HEAD_request(httpserver):
 #     assert resp.status_code == 200
 
 
+def test_POST_request_data(httpserver):
+    headers = {"Content-type": "text/plain"}
+    httpserver.serve_content('TEST!', store_request_data=True)
+    requests.post(httpserver.url, data=b'testdata', headers=headers)
+    assert httpserver.requests[-1].data == b'testdata'
+
+
+def test_POST_request_json(httpserver):
+    test_data = [{"test": "mock"}, 42]
+    headers = {"Content-type": "application/json"}
+    httpserver.serve_content('TEST!', store_request_data=True)
+    requests.post(httpserver.url, json=test_data, headers=headers)
+    assert httpserver.requests[-1].json == test_data
+
+
```

Comment:
That was a good point. I stumbled upon what probably was a race condition where the stream was still readable.
Sleeping just 0.1 seconds fixed it. 

For the tests I decided to close the stream manually for these cases instead of sleeping.